src/evm-block-extractor/src/blocks_reader.rs

```diff
@@ -0,0 +1,71 @@
+use std::fs::File;
+use std::path::Path;
+
+use zip::read::ZipArchive;
+
+pub struct BlocksReader {
+    reader: ZipArchive<File>,
+}
+
+impl BlocksReader {
+    pub fn open(path: &Path) -> anyhow::Result<Self> {
+        let file = File::open(path)?;
+        let reader = ZipArchive::new(file)?;
+
+        Ok(Self { reader })
+    }
+
+    pub fn get_last_block_number(&mut self) -> anyhow::Result<u64> {
+        if self.reader.is_empty() {
+            anyhow::bail!("No blocks found");
+        }
+        let last_block_file = self.reader.by_index(self.reader.len() - 1)?;
+        let last_block_number = Self::block_number_from_file_name(last_block_file.name())?;
+
+        Ok(last_block_number)
+    }
+
+    fn block_number_from_file_name(file_name: &str) -> anyhow::Result<u64> {
+        let prefix = if file_name.starts_with("block") {
```

Comment:
if it does not start with "block" then it is not a block file so it should return an error